File: WinRT/FileTransfer/FileTransfer.Shared/TransferClasses.cs
using System.Runtime.Serialization;
namespace FileTransfer
{
    [DataContract]
    public class Request
    {
        [DataMember(Name = Strings.JSON_REQUEST_TYPE)]
        public string Type;
        [DataMember(Name = Strings.JSON_REQUEST_DATA)]
        public string Data;
        [DataMember(Name = Strings.JSON_GUID)]
        public string SenderGuid;
    }
    [DataContract]
    public class PreSendRequest
    {
        [DataMember(Name = Strings.JSON_FILE_NAME)]
        public string FileName;
        [DataMember(Name = Strings.JSON_PUBLICKEY)]
        public PublicKey key;
    }
    [DataContract]
    public class PreSendRequestNoCrypt
    {
        [DataMember(Name = Strings.JSON_FILE_NAME)]
        public string FileName;
    }
    [DataContract]
    public class PublicKey
    {
        [DataMember(Name=Strings.JSON_PUBLICKEY)]
        public string Key;
        [DataMember(Name=Strings.JSON_SIGNATURE)]
        public string Signature;
        public bool VerifySignature(string secret)
        {
            return Cryptography.Signatures.VerifySignature(secret, Key, Signature);
        }
        public PublicKey(string key, string secret)
        {
            Key = key;
            Signature = Cryptography.Signatures.GenerateSignature(secret, key);
        }
        public PublicKey()
        {

        }
    }
    [DataContract]
    public class PairingRequest
    {
        [DataMember(Name = Strings.JSON_NAME)]
        public string Name;
        [DataMember(Name = Strings.JSON_GUID)]
        public string Guid;
        [DataMember(Name = Strings.JSON_PUBLICKEY)]
        public PublicKey PublicKey;
    }
    [DataContract]
    public class File
    {
        [DataMember(Name = Strings.JSON_FILE_CONTENTS)]
        public string Contents;
        [DataMember(Name=Strings.JSON_FILE_NAME)]
        public string FileName;
        public File()
        {

        }
        public File(string fileName, byte[] fileContents, byte[] aesKey)
        {
            string filePrivateKey = Cryptography.DiffieHellman.generate_DH_Private();
            Contents = Cryptography.AES.Encrypt(fileContents, aesKey);
            FileName = fileName;

        }
        public File(string fileName, byte[] fileContents)
        {
            Contents = Cryptography.TextManipulation.Encode64(fileContents);
            FileName = fileName;
        }
        public byte[] GetContents(byte[] aesKey)
        {
            return Cryptography.AES.Decrypt(Contents, aesKey);
        }
        public byte[] GetContents()
        {
            return Cryptography.TextManipulation.Decode64(Contents);
        }
    }
}

File: Python/transferclasses.py
from cryptography import generateSignature, verifySignature, generateDHPrivate, calculateDHPublic, calculateDHAES, \
    AESEncrypt, AESDecrypt
from strings import *


class Request:
    jsonKeys = {"type": JSON_REQUEST_TYPE,
                "data": JSON_REQUEST_DATA,
                "guid": JSON_GUID}

    def __init__(self, type, data, guid):
        self.type = type
        self.data = data
        self.guid = guid


class PublicKey:
    jsonKeys = {"key": JSON_PUBLICKEY,
                "signature": JSON_SIGNATURE}

    def __init__(self, key, secret):
        self.key = key
        self.signature = generateSignature(self.key, secret)

    def verifySignature(self, secret):
        return verifySignature(self.key, secret, self.signature)


class Peer:
    jsonKeys = {"guid": JSON_GUID,
                "name": JSON_NAME,
                "myPrivateKey": JSON_PRIVATEKEY,
                "publicKey": JSON_PUBLICKEY,
                "sharedPassword": "sharedPassword"}

    def __new__(cls, *args, **kwargs):
        newObject = object.__new__(Peer)
        newObject.lastKnownIP = None
        newObject.myPrivateKey = None
        newObject.publicKey = None
        newObject.sharedPassword = None
        newObject.lastKnownIP = None
        return newObject

    def __init__(self, guid, name):
        self.guid = guid
        self.name = name
        self.myPrivateKey = None
        self.publicKey = None
        self.sharedPassword = None
        self.lastKnownIP = None

    def pair(self, myPrivateKey, publicKey, sharedPassword):
        self.myPrivateKey = myPrivateKey
        self.publicKey = publicKey
        self.sharedPassword = sharedPassword


class PairingRequest:
    jsonKeys = {"name": JSON_NAME,
                "guid": JSON_GUID,
                "publicKey": (JSON_PUBLICKEY, PublicKey)}

    def __init__(self, name, guid, publicKey):
        self.name = name
        self.guid = guid
        self.publicKey = publicKey


class File:
    jsonKeys = {"contents": JSON_FILE_CONTENTS,
                "fileName": JSON_FILE_NAME}

    def __init__(self, fileName, fileContents, aesKey):
        self.contents = AESEncrypt(fileContents, aesKey)
        self.fileName = fileName

    def getContents(self, aesKey):
        return AESDecrypt(self.contents, aesKey, False)


class PreSendRequest:
    jsonKeys = {"fileName": JSON_FILE_NAME,
                "publicKey": (JSON_PUBLICKEY, PublicKey)}

    def __init__(self, fileName, publicKey):
        self.publicKey = publicKey
        self.fileName = fileName
